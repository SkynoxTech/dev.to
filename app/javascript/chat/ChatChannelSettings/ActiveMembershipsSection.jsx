import { h } from 'preact';
import PropTypes from 'prop-types';
import Membership from './Membership';

const ActiveMembershipSection = ({
  activeMemberships,
  removeMembership,
  currentMembershipRole,
  toggelScreens,
}) => {
  const RenderActiveMembershipManager = () => {
    toggelScreens();
  };

  const activeMembershipList = activeMemberships.slice(0, 4);

  return (
    <div
      data-testid="active-memberships"
      className="p-4 grid gap-2 crayons-card mb-4"
      data-active-count={activeMemberships ? activeMemberships.length : 0}
    >
      <h3 className="mb-2 active_members">Members</h3>
      {activeMembershipList && activeMembershipList.length > 0
        ? activeMembershipList.map((activeMembership) => (
          <Membership
            membership={activeMembership}
            removeMembership={removeMembership}
            membershipType="active"
            currentMembershipRole={currentMembershipRole}
            className="active-member"
          />
          ))
        : null}
      <div className="row view-membership-btn">
        <button
          className="crayons-btn align-center crayons-btn--s view-all-memberships"
          onClick={() => RenderActiveMembershipManager()}
          type="button"
        >
          View All
        </button>
      </div>
    </div>
  );
};

ActiveMembershipSection.propTypes = {
  activeMemberships: PropTypes.arrayOf(
    PropTypes.shape({
      name: PropTypes.string.isRequired,
      membership_id: PropTypes.number.isRequired,
      user_id: PropTypes.number.isRequired,
      role: PropTypes.string.isRequired,
      image: PropTypes.string.isRequired,
      username: PropTypes.string.isRequired,
      status: PropTypes.string.isRequired,
    }),
  ).isRequired,
  removeMembership: PropTypes.func.isRequired,
  currentMembershipRole: PropTypes.string.isRequired,
  toggelScreens: PropTypes.func.isRequired,
};

export default ActiveMembershipSection;
