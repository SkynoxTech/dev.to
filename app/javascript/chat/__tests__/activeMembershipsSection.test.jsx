import { h } from 'preact';
import render from 'preact-render-to-json';
import { shallow } from 'preact-render-spy';
import ActiveMembershipsSection from '../ChatChannelSettings/ActiveMembershipsSection';

const data = {
  activeMemberships: [],
  currentMembershipRole: 'mod'
}

const getActiveMembershipsSection = (membershipData) => (
  <ActiveMembershipsSection 
    activeMemberships={membershipData.activeMemberships}
    currentMembershipRole={membershipData.currentMembershipRole}
  />
)

describe('<ActiveMembershipsSection />', () => {
  it("should render and test snapshot", () => {
    const tree = render(getActiveMembershipsSection(data));
    expect(tree).toMatchSnapshot();
  })

  it("should have the elements", () => {
    const context = shallow(getActiveMembershipsSection(data));

    expect(context.find('.active_members').exists()).toEqual(true)
  })
})