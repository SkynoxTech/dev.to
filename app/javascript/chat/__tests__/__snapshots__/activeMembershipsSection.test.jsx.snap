// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ActiveMembershipsSection /> should render and test snapshot 1`] = `
<div
  class="p-4 grid gap-2 crayons-card mb-4"
>
  <h3
    class="mb-2 active_members"
  >
    Members
  </h3>
  <div
    class="row view-membership-btn"
  >
    <button
      class="crayons-btn align-center crayons-btn--s view-all-memberships"
      onClick={[Function]}
      type="button"
    >
      View All
    </button>
  </div>
</div>
`;
